service/src/main/kotlin/no/nav/su/se/bakover/service/behandling/BehandlingServiceImpl.kt

```diff
@@ -0,0 +1,171 @@
+        behandlingId: UUID,
+        fraOgMed: LocalDate,
+        tilOgMed: LocalDate,
+        fradrag: List<Fradrag>
+    ): Behandling {
+        beregningRepo.slettBeregningForBehandling(behandlingId)
+        val beregnet = behandlingRepo.hentBehandling(behandlingId)!!.opprettBeregning(fraOgMed, tilOgMed, fradrag)
+        beregningRepo.opprettBeregningForBehandling(behandlingId, beregnet.beregning()!!)
+        behandlingRepo.oppdaterBehandlingStatus(behandlingId, beregnet.status())
+        return behandlingRepo.hentBehandling(behandlingId)!!
+    }
+
+    override fun simuler(behandlingId: UUID): Either<SimuleringFeilet, Behandling> {
+        val behandling = behandlingRepo.hentBehandling(behandlingId)!!
+        val sak = sakService.hentSak(behandling.sakId).fold(
+            { throw RuntimeException("Kunne ikke finne sak") },
+            { it }
+        )
+        val oppdrag = oppdragRepo.hentOppdrag(behandling.sakId)!!
+        val utbetaling = oppdrag.genererUtbetaling(behandling.beregning()!!, sak.fnr)
+        val utbetalingTilSimulering = NyUtbetaling(oppdrag, utbetaling, Attestant("SU"))
+        return simuleringClient.simulerUtbetaling(utbetalingTilSimulering)
+            .map {
+                behandling.utbetaling()?.let { utbetalingService.slettUtbetaling(it) }
+                utbetalingService.opprettUtbetaling(oppdrag.id, utbetaling)
+                utbetalingService.addSimulering(utbetaling.id, it)
+                behandlingRepo.leggTilUtbetaling(behandlingId, utbetaling.id)
+                val oppdatert = behandlingRepo.hentBehandling(behandlingId)!!
+                oppdatert.simuler(utbetaling) // TODO just to push to correct state
+                behandlingRepo.oppdaterBehandlingStatus(behandling.id, oppdatert.status())
+                return behandlingRepo.hentBehandling(behandlingId)!!.right()
+            }
+    }
+
+    override fun sendTilAttestering(
+        behandlingId: UUID,
+        aktørId: AktørId,
+        saksbehandler: Saksbehandler
+    ): Either<KunneIkkeOppretteOppgave, Behandling> {
+        val behandling = behandlingRepo.hentBehandling(behandlingId)!!
+        return oppgaveClient.opprettOppgave(
+            OppgaveConfig.Attestering(
+                behandling.sakId.toString(),
+                aktørId = aktørId
+            )
+        ).map {
+            behandlingRepo.settSaksbehandler(behandlingId, saksbehandler)
+            val oppdatert = behandlingRepo.hentBehandling(behandlingId)!!
+            oppdatert.sendTilAttestering(aktørId, saksbehandler)
+            behandlingRepo.oppdaterBehandlingStatus(behandlingId, oppdatert.status())
+            behandlingRepo.hentBehandling(behandlingId)!!
+        }
+    }
+
+    override fun iverksett(behandlingId: UUID, attestant: Attestant): Either<Behandling.IverksettFeil, Behandling> {
+        val behandling = behandlingRepo.hentBehandling(behandlingId)!!
+        val utbetaling = behandling.utbetaling()!!
```

Comment:
Burde kalle iverksett før man antar at det finnes en utbetaling

Reply:
Ja. Bør legge inn et kall til aktuell metode på domeneobjektet før alle operasjoner på dette service-interfacet tenker jeg. Dette for å sørge for at tilstandene oppfører seg som før.